kernel/src/process.rs

```diff
@@ -103,7 +103,7 @@ pub fn schedule(callback: FunctionCall, appid: AppId) -> bool {
 /// app owns and can write to. This includes the app's code and data and any
 /// padding at the end of the app. It does not include the TBF header, or any
 /// space that the kernel is using for any potential bookkeeping.
-pub fn get_editable_flash_range(app_idx: usize) -> (usize, usize) {
+crate fn get_editable_flash_range(app_idx: usize) -> (usize, usize) {
```

Comment:
I suspect this particular method will be useful for capsules that access flash. But if there's another pub method that effectively exports it (with more checks?) that could be OK.

Reply:
Agreed that this is potentially useful functionality to expose.

Doing it with a raw `usize` for the app id isn't _definitely_ not stable (since we have long had plans to change the underlying format and meaning of an AppId, and probably will always want to keep that opaque). It also likely should be a privileged operation of some sort (e.g. requiring a capability).

Reply:
We do:

https://github.com/tock/tock/blob/f638195c8824cbf314945fc27c36caa203ad44a5/kernel/src/callback.rs#L37-L39